frame/merkle-mountain-range/src/tests.rs

```diff
@@ -95,95 +117,93 @@ fn should_start_empty() {
-		assert_eq!(crate::Nodes::<Test>::get(0), None);
+		assert_eq!(crate::Peaks::<Test>::get(0), None);
 
 		// when
-		let weight = new_block();
+		let (_, _, weight) = new_block();
 
 		// then
 		assert_eq!(crate::NumberOfLeaves::<Test>::get(), 1);
 		assert_eq!(
-			crate::Nodes::<Test>::get(0),
-			Some(hex("4320435e8c3318562dba60116bdbcc0b82ffcecb9bb39aae3300cfda3ad0b8b0"))
+			crate::Peaks::<Test>::get(0),
+			Some(hex("c51a80dfad5ad167c7001f581065bea8db70b04aeca50eedf908cbb3ff859d0b"))
 		);
 		assert_eq!(
 			crate::RootHash::<Test>::get(),
-			hex("4320435e8c3318562dba60116bdbcc0b82ffcecb9bb39aae3300cfda3ad0b8b0")
+			hex("c51a80dfad5ad167c7001f581065bea8db70b04aeca50eedf908cbb3ff859d0b")
 		);
 		assert!(weight != Weight::zero());
 	});
 }
 
 #[test]
-fn should_append_to_mmr_when_on_initialize_is_called() {
+fn should_append_to_mmr_after_on_initialize() {
 	let _ = env_logger::try_init();
 	let mut ext = new_test_ext();
-	let (parent_b1, parent_b2) = ext.execute_with(|| {
+	let (hash1, hash2) = ext.execute_with(|| {
 		// when
-		new_block();
-		let parent_b1 = <frame_system::Pallet<Test>>::parent_hash();
+		let (_, hash1, _) = push_block_with_offchain();
 
 		// then
 		assert_eq!(crate::NumberOfLeaves::<Test>::get(), 1);
 		assert_eq!(
 			(
-				crate::Nodes::<Test>::get(0),
-				crate::Nodes::<Test>::get(1),
+				crate::Peaks::<Test>::get(0),
+				crate::Peaks::<Test>::get(1),
 				crate::RootHash::<Test>::get(),
 			),
 			(
-				Some(hex("4320435e8c3318562dba60116bdbcc0b82ffcecb9bb39aae3300cfda3ad0b8b0")),
+				Some(hex("c51a80dfad5ad167c7001f581065bea8db70b04aeca50eedf908cbb3ff859d0b")),
 				None,
-				hex("0x4320435e8c3318562dba60116bdbcc0b82ffcecb9bb39aae3300cfda3ad0b8b0"),
+				hex("0xc51a80dfad5ad167c7001f581065bea8db70b04aeca50eedf908cbb3ff859d0b"),
 			)
 		);
 
 		// when
-		new_block();
-		let parent_b2 = <frame_system::Pallet<Test>>::parent_hash();
+		let (_, hash2, _) = push_block_with_offchain();
 
 		// then
 		assert_eq!(crate::NumberOfLeaves::<Test>::get(), 2);
 		let peaks = peaks_from_leaves_count(2);
 		assert_eq!(peaks, vec![2]);
 		assert_eq!(
 			(
-				crate::Nodes::<Test>::get(0),
-				crate::Nodes::<Test>::get(1),
-				crate::Nodes::<Test>::get(2),
-				crate::Nodes::<Test>::get(3),
+				crate::Peaks::<Test>::get(0),
+				crate::Peaks::<Test>::get(1),
+				crate::Peaks::<Test>::get(2),
+				crate::Peaks::<Test>::get(3),
 				crate::RootHash::<Test>::get(),
 			),
 			(
 				None,
 				None,
-				Some(hex("672c04a9cd05a644789d769daa552d35d8de7c33129f8a7cbf49e595234c4854")),
+				Some(hex("843450e593a9103c9fe8c0a0276756a90077108b9aa32d6a15c1b77f264b097e")),
 				None,
-				hex("672c04a9cd05a644789d769daa552d35d8de7c33129f8a7cbf49e595234c4854"),
+				hex("843450e593a9103c9fe8c0a0276756a90077108b9aa32d6a15c1b77f264b097e"),
 			)
 		);
 
-		(parent_b1, parent_b2)
+		(hash1, hash2)
 	});
 	// make sure the leaves end up in the offchain DB
 	ext.persist_offchain_overlay();
```

Comment:
all of the `persist_offchain_overlay()` calls in the mmr tests become superfluous with the changes in this PR, so I'd remove them.